File: src/main/java/com/sat/member/domain/Member.java
package com.sat.member.domain;

import jakarta.persistence.Embedded;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;


@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Entity
public class Member {

    @Id @EmbeddedId
    private MemberId id;

    @Embedded
    private Name name;

    public Member(String name) {
        this(null, new Name(name));
    }

    public Member(MemberId id, String name) {
        this(id, new Name(name));
    }

    public Member(MemberId id, Name name) {
        this.id = id;
        this.name = name;
    }

    public String getId() {
        return id.getId();
    }
}

File: src/main/kotlin/com/sat/member/domain/Member.kt
package com.sat.member.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class Member(
    val name: String,
    val nickname: String,
    val email: String,
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
) {
}
